### PR TITLE
#18 Chore: Setup Continuous Integration & Quality Assurance pipeline

### PR DESCRIPTION
**Pull request by benshyaka**
#### What does this PR do?

- Configuring testing environment
- Configuring continous integration and deployement 
- Adds Github actions badge to the repo

#### Description of Task to be completed?

- Codes are continously tested before the deployment 
- Github action badge are displayed
- Codeclimate to ensure the code analysis and maintainability checking.

#### How should this be manually tested?

- Clone the repo

- checkout to branch `ch-setup-integrations-18`

- Run `npm run install` to install the dependencies

- Run `npm run coverage` to run testing coverage report

- Run `npm run test` to run tests

- Check the ReadMe of the repo to confirm the github actions badges are displayed

#### What are the relevant Pivotal Tracker/Trello stories?

- https://trello.com/c/WgAyuZBs/18-setup-continuous-integration-quality-assurance-pipeline

#### Screenshots (if appropriate)

-  N/A

#### Questions:

-  N/A

### CODE REVIEW

**Review by PrinceNiyonshuti — APPROVED** [review]
looks good to me

File: package.json [hunk 1]
```diff
@@ -7,9 +7,9 @@
     "scripts": {
         "start:dev": "cross-env NODE_ENV=dev nodemon --exec babel-node src/index.js",
         "start": "cross-env NODE_ENV=prod node dist/index.js",
-        "test": "cross-env NODE_ENV=test nyc mocha --require @babel/register ./src/tests/ --recursive  --exit --timeout 30000",
+        "test": "cross-env NODE_ENV=test nyc mocha './src/**/*.test.js' --require @babel/register --recursive  --exit --timeout 30000",
         "build": "babel src --out-dir dist",
-        "coverage": "nyc --reporter=lcov --reporter=text yarn test"
```
Comment (Hodal66) [inline package.json:12]: Hello Why did you remove this reporter at 12 line in package.json

Comment (thepatrickniyo) [reply package.json:12]: @Hodal66 I think he replaced it with `"coverage": "nyc npm run test"` at it will still log our test coverage.

**Review by thepatrickniyo — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (Hodal66) [reply package.json:12]: OK nice!!
